src/ui/public/query_manager/query_manager.js

```diff
@@ -1,79 +1,24 @@
-import { FilterBarLibExtractTimeFilterProvider } from 'ui/filter_bar/lib/extract_time_filter';
-import { FilterBarLibChangeTimeFilterProvider } from 'ui/filter_bar/lib/change_time_filter';
-import { toKueryExpression, fromKueryExpression, nodeTypes, filterToKueryAST } from 'ui/kuery';
 
-export function QueryManagerProvider(Private) {
-  const filterManager = Private(FilterManagerProvider);
-  const mapAndFlattenFilters = Private(FilterBarLibMapAndFlattenFiltersProvider);
-  const extractTimeFilter = Private(FilterBarLibExtractTimeFilterProvider);
-  const changeTimeFilter = Private(FilterBarLibChangeTimeFilterProvider);
+export function queryManagerFactory(state) {
 
-  return function (state) {
-
-    function add(field, values = [], operation, index) {
-      const fieldName = _.isObject(field) ? field.name : field;
-
-      if (!Array.isArray(values)) {
-        values = [values];
-      }
-
-      if (state.query.language === 'lucene') {
-        filterManager.add(field, values, operation, index);
-      }
-
-      if (state.query.language === 'kuery') {
-        const negate = operation === '-';
-        const isExistsQuery = fieldName === '_exists_';
-
-        const newQueries = values.map((value) => {
-          const newQuery = isExistsQuery
-            ? nodeTypes.function.buildNode('exists', value)
-            : nodeTypes.function.buildNode('is', fieldName, value);
-
-          return negate ? nodeTypes.function.buildNode('not', newQuery) : newQuery;
-        });
-
-        const allQueries = _.isEmpty(state.query.query)
-          ? newQueries
-          : [fromKueryExpression(state.query.query), ...newQueries];
-
-        state.query = {
-          query: toKueryExpression(nodeTypes.function.buildNode('and', allQueries, 'implicit')),
-          language: 'kuery'
-        };
-      }
-    }
+  function getQuery() {
+    return {
+      ...state.query
+    };
+  }
 
-    async function addLegacyFilter(filter) {
-      // The filter_bar directive currently handles filter creation when lucene is the selected language,
-      // so we only handle updating the kuery AST here.
-      if (state.query.language === 'kuery') {
-        const timeFilter = await extractTimeFilter([filter]);
-        if (timeFilter) {
-          changeTimeFilter(timeFilter);
-        }
-        else {
-          const [ mappedFilter ] = await mapAndFlattenFilters([filter]);
-          const newQuery = filterToKueryAST(mappedFilter);
-          const allQueries = _.isEmpty(state.query.query)
-            ? [newQuery]
-            : [fromKueryExpression(state.query.query), newQuery];
+  function updateQuery(newQuery) {
+    state.query = newQuery;
 
-          state.query = {
-            query: toKueryExpression(nodeTypes.function.buildNode('and', allQueries, 'implicit')),
-            language: 'kuery'
-          };
-        }
-      }
+    if (_.isFunction(state.save)) {
+      state.save();
```

Comment:
This is a hack to ensure Dashboard reacts to changes made by visualizations via the query manager. Dashboard only watches for appState changes that have been saved, unlike Discover and Visualize.